src/main/java/com/tabnine/userSettings/AppSettingsState.kt

```diff
@@ -27,6 +29,11 @@ class AppSettingsState : PersistentStateComponent<AppSettingsState?> {
     var autoImportEnabled: Boolean = true
     var binariesFolderOverride: String = ""
     var cloud2Url: String = ""
+        set(value) {
```

Comment:
im not in favor of performing these operations here, please move this logic to [here](https://github.com/codota/tabnine-intellij/blob/45de31701d23f424aa360d06aa0dcf73fcf9fc23/src/main/java/com/tabnine/userSettings/AppSettingsConfigurable.kt#L47) or some other method(s) in this file

Reply:
but... that what you actually have setters for...

Reply:
Setters are for lightweight operations of coupled states, and I think its extra clear in Kotlin where the syntactic sugaring for setters is literally the `=` operation - you can't expect heavy operations to be performed on assignment, right?

but ` UpdateSettings.getInstance().storedPluginHosts` is a more serious operation - it includes `ServiceManager.getService`, which can be IDE/version specific, and it includes multiple file reads which can not only fail but slow down the process or something.

Invoking all the above on a setter can be misleading, especially when it's being called not only from the UI component.

Reply:
The problem with where you suggested - it's will call only when it changed via the UI, but I want the effect to be called where ever it changed, not matter if via code or via UI.